File: lacoRecursivo.cpp
#include <iostream>
using namespace std;

int laco(int *lista, int total, int passos=1, int i=0){
    if (i >= total){
        cout<<"Preenchida";
    }
    else{
        lista[i] = i;
        return laco(lista,total,passos, i + passos);
    }
};

int main(){
    int lista[10];

    laco(lista,10);
    for(int j = 0; j < 10; j++)
        cout<<" | " <<lista[j];

    return 0;
}

File: lacoRecursivo.py
def laco(lista,total,p=1, i=0):
    if i >= total:
        pass
    else:
        lista.append(i)
        laco(lista,total,p,i+p)
        return lista

"O algoritmo precisa recebar a lista a qual voce deseja intera sobre ela e a quantidade de itens que voce deseja preencher, e os passos caso queira"

lista = list()
laco(lista,13)
print(lista)

"Caso não defina os passos que o algoritmo deva seguir, o algoritmo toma como padrão avança de 1 em 1"